drivers/memory_window/dynamic_slots.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "dynamic_slots.h"
+
+#if CONFIG_MM_DRV_INTEL_ADSP_MTL_TLB
+#include <zephyr/drivers/mm/system_mm.h>
+#endif /* CONFIG_MM_DRV_INTEL_ADSP_MTL_TLB */
+#include <zephyr/arch/xtensa/cache.h>
+
+#if CONFIG_MM_DRV_INTEL_ADSP_MTL_TLB
+static uint8_t tlb[MEMORY_WINDOW_SLOTS_COUNT];
+#endif /* CONFIG_MM_DRV_INTEL_ADSP_MTL_TLB */
+
+uint32_t dynamic_slots_init(const struct device *dev)
+{
+	uint32_t buf_size = MEMORY_WINDOW_SLOT_SIZE * MEMORY_WINDOW_SLOTS_COUNT;
+	const struct mem_win_config *config = dev->config;
+
+	uint32_t *virt_buf = (uint32_t __sparse_cache *)
+		arch_xtensa_cached_ptr((void *)(config->mem_base));
```

Comment:
`arch_xtensa_cached_ptr()` returns a pointer from `__sparse_cache` address space, so here you want to remove it instead of enforcing it again:
```
uint32_t *virt_buf = (__sparse_force uint32_t *)
		arch_xtensa_cached_ptr((void *)(config->mem_base));
```